spec/gilded_rose_spec.js

```diff
@@ -1,7 +1,267 @@
+'use strict'
+
+function resetItems(){
+  items = []
+}
+
+function populateItems(){
+  items.push(new Item('+5 Dexterity Vest', 10, 20));
+  items.push(new Item('Aged Brie', 2, 0));
+  items.push(new Item('Elixir of the Mongoose', 5, 7));
+  items.push(new Item('Sulfuras, Hand of Ragnaros', 0, 80));
+  items.push(new Item('Backstage passes to a TAFKAL80ETC concert', 15, 20));
+  items.push(new Item('Conjured Mana Cake', 3, 20));
+}
+
 describe("Gilded Rose", function() {
+  describe("items have a sell-in value", () => {
+    it("should initialize a quality", function() {
+      resetItems()
+      populateItems()
+
+      expect(items[0].sell_in).toEqual(10)
+      expect(items[1].sell_in).toEqual(2)
+      })
+    })
+  describe("items have a quality value", () => {
+    it("should initialize a quality", () => {
+      resetItems()
+      populateItems()
+
+      expect(items[4].quality).toEqual(20)
+      expect(items[5].quality).toEqual(20)
+    })
+  })
+  describe("Dexterity Vest", () => {
+    it("sell_in value decrease by 1 each day", () => {
+      resetItems()
+      populateItems()
+      update_quality()
+
+      expect(items[0].sell_in).toEqual(9)
```

Comment:
What is item[0]? Sure, it's dexterity because it is in the describe six line above so there is small chance I will see that, but what about later tests?

Consider:
```
var dexterityItem = new Item('dexterity', 9, 9)
items = [dexterityItem]
update_quality()
expect(dexterityItem.sell_in).toEqual(9
```
this test is isoated from other tests and the reader knows the beginning and ending state.